bazel: rename @cockroach to @com_github_cockroachdb_cockroach

otan opened the pull request:
This further complies with how other repos may import CockroachDB using
bazel.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80342)
<!-- Reviewable:end -->

rickystewart reviewed (approved):
Let's just make sure CI doesn't complain before landing this.

otan committed:
bazel: rename @cockroach to @com_github_cockroachdb_cockroach

This further complies with how other repos may import CockroachDB using
bazel.

Release note: None

otan commented:
bazel failure is a flake on logictest

bors r=rickystewart
